content/blog/2024/11/how-to-create-an-esp-idf-component/index.md

```diff
@@ -0,0 +1,483 @@
+
+You can add the project components from the ESP registry, from third-party sources, or you can choose to create your own components. This article covers the project components you can create yourself and add to your project.
+
+
+## Writing the ESP-IDF component
+
+Frequently, placing self-contained parts of code in a separate component not only helps to keep the application code cleaner, but also simplifies development and maintenance. To get started with the component development process, we need to understand the basic structure and how the component will interact with the application.
+
+
+As covered in the article What is the ESP Component Registry, a component can be either created directly in the application or in a separated project and then added to the application.
+
+
+{{< article link="/blog/2024/10/what-is-the-esp-registry/" >}}
+
+When choosing whether to create a component in a separate project, consider if you plan to share the component with other applications. If yes, the best approach is to create the component in a new project that will contain just the component and examples if needed.
+
+
+{{< alert >}}
+This article will not cover how to publish the component in the ESP Component Registry. It will be covered in the next article of this series.
+{{< /alert >}}
+
+
+Now it's time to start with the new component. For this article, we will use an I2C temperature and humidity sensor.
+
+### The sensor: SHTC3
+
+For this new component, the sensor will be the [Sensirion SHTC3](https://sensirion.com/products/catalog/SHTC3). This is a I2C temperature and humidity sensor with a typical accuracy of ±2 %RH and ±0.2 °C. To know more about the specifications, please visit the official [product page](https://sensirion.com/products/catalog/SHTC3).
+
+If you don't have this sensor, you can change to any other I2C device. All the information you might need for the new sensor should be provided by the manufacturer, including the device address, registers, etc.
+
+### The Board: ESP32-C3-DevKit-RUST-1
+
+We chose the sensor SHTC3, because it can be found on the [ESP32-C3-DevKit-RUST-1](https://github.com/esp-rs/esp-rust-board/tree/v1.2) development board. That is why, this is the board that we will use for developing the component.
+
+
+The **ESP32-C3-DevKit-RUST-1** development board has 2 sensors connected to the I2C peripheral on the following GPIOs:
+
+| Signal     | GPIO        |
+|------------|-------------|
+| SDA        | GPIO10      |
+| SCL        | GPIO8       |
+
+All the project files for this board are available on GitHub.
+
+
+{{< github repo="esp-rs/esp-rust-board" >}}
+
+### Creating the component
+
+Before creating the component, we will need to create a new project that the component will be part of. If you have already the project, you can skip this step and jump to the component creation.

```

Comment:
It's not always the case, and we can have standalone components outside a project. I would suggest rephrasing, saying that in this tutorial we are going to create a component inside the project.

Reply:
This is outdated. Please check the new sentence:

`To create a new component, we will use a project as the starting point. This project will be used to develop and test the component. However, you can also create the component as a stand-alone without using a project.`